### PR TITLE
Update docs styling guidelines for code blocks

### PR DESCRIPTION
**Pull request by plumbis**
This adds guidelines around how code blocks should be written in the documentation.

In summary:
* All code should be in a code fence `` ``` `` unless it's a command inside a sentence.
* Code fences must provide highlighting hints
* Highlighting hints should optimize readability over technical correctness. (i.e., a shell output that is 99% yaml is better off highlighting as yaml than shell)

I highly recommend any reviewers read this on a [local render](https://docs.crossplane.io/master/contributing/docs/#build-the-crossplane-documentation) of the docs and not just as markdown.

Specifically tagging @negz for review as docs styling was something we had discussed in #3396

### CODE REVIEW

File: docs/contributing/docs.md [hunk 1]
```diff
@@ -142,6 +142,68 @@ definitions](https://github.com/upbound/vale) for style guidelines.
...
+three backticks (` ``` `) for
+all command examples and outputs.
+
+````markdown
+```
+this is a code block
+```
+````
+
+Only use single backtick (`` ` ``) for commands used in a sentence.
```
Comment (hasheddan) [inline docs/contributing/docs.md:159]: nitpick:
```suggestion
Only use a single backtick (`` ` ``) for commands used in a sentence. 
```

File: docs/contributing/docs.md [hunk 2]
```diff
@@ -142,6 +142,68 @@ definitions](https://github.com/upbound/vale) for style guidelines.
...
 
+### Code examples
+
+#### Fenced code blocks
+Use Markdown [fenced code
+blocks](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks) with
+three backticks (` ``` `) for
+all command examples and outputs.
+
+````markdown
```
Comment (hasheddan) [inline docs/contributing/docs.md:153]: TIL about 4 backtick syntax 👀

**Review by hasheddan — APPROVED** [review]
LGTM but will wait for @negz approval since specifically mentioned 👍🏻

File: docs/contributing/docs.md (hunk 2 above)
Comment (plumbis) [reply docs/contributing/docs.md:153]: "Hey Markdown, how do I escape backticks? 
"MOAR BACKTICKS!"

**Review by negz — APPROVED** [review]

**Pull request merged**